pkg/controllers/provisioning/volumetopology.go

```diff
@@ -154,19 +155,43 @@ func (v *VolumeTopology) validatePersistentVolumeClaims(ctx context.Context, pod
 			// may not have a PVC
 			if pvc == nil {
 				continue
-
 			}
+
 			storageClassName = pvc.Spec.StorageClassName
+			volumeName = pvc.Spec.VolumeName
 		} else if volume.Ephemeral != nil {
 			storageClassName = volume.Ephemeral.VolumeClaimTemplate.Spec.StorageClassName
 		}
 
-		// we have a storage class name, so ensure that it exists
-		if storageClassName != nil && ptr.StringValue(storageClassName) != "" {
-			storageClass := &storagev1.StorageClass{}
-			if err := v.kubeClient.Get(ctx, types.NamespacedName{Name: ptr.StringValue(storageClassName)}, storageClass); err != nil {
-				return err
-			}
+		err := v.validateStorageClass(ctx, storageClassName)
+		if err != nil {
+			return err
+		}
+		err = v.validateVolume(ctx, volumeName)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (v *VolumeTopology) validateVolume(ctx context.Context, volumeName string) error {
+	// we have a volume name, so ensure that it exists
+	if volumeName != "" {
+		pv := &v1.PersistentVolume{}
+		if err := v.kubeClient.Get(ctx, types.NamespacedName{Name: volumeName}, pv); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (v *VolumeTopology) validateStorageClass(ctx context.Context, storageClassName *string) error {
+	// we have a storage class name, so ensure that it exists
+	if storageClassName != nil && ptr.StringValue(storageClassName) != "" {
```

Comment:
Prefer `lo.FromPtr`. I thin we are hoping to pull out `ptr` calls that we have written ourselves.

Also, I think this can be collapsed since `StringValue` will evaluate to `""` when its nil

Reply:
We shouldn't be using generics for such simple stuff unless it offers some strong benefit.  It's over 30x slower in this case:
```
BenchmarkLoPtr-12    	142887817	         8.386 ns/op
BenchmarkFromStr-12    	1000000000	         0.2556 ns/op
```